### ISSUE TITLE
PersistentStateScopeResourceDescriptionsTest fails on photon

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#494]
```-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.eclipse.xtext.ui.tests.resource.PersistentStateScopeResourceDescriptionsTest
Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.907 sec <<< FAILURE! - in org.eclipse.xtext.ui.tests.resource.PersistentStateScopeResourceDescriptionsTest
testLiveScope2(org.eclipse.xtext.ui.tests.resource.PersistentStateScopeResourceDescriptionsTest)  Time elapsed: 0.818 sec
testDirtyStateAware(org.eclipse.xtext.ui.tests.resource.PersistentStateScopeResourceDescriptionsTest)  Time elapsed: 0.033 sec  <<< ERROR!
org.eclipse.emf.ecore.resource.impl.ResourceSetImpl$1DiagnosticWrappedException: org.eclipse.core.internal.resources.ResourceException: Resource '/test/test.testlanguage' does not exist.
	at org.eclipse.core.internal.resources.ResourceException.provideStackTrace(ResourceException.java:39)
	at org.eclipse.core.internal.resources.ResourceException.<init>(ResourceException.java:35)
	at org.eclipse.core.internal.resources.Resource.checkExists(Resource.java:335)
	at org.eclipse.core.internal.resources.Resource.checkAccessible(Resource.java:209)
	at org.eclipse.core.internal.resources.File.getContents(File.java:275)
	at org.eclipse.core.internal.resources.File.getContents(File.java:268)
	at org.eclipse.emf.ecore.resource.impl.PlatformResourceURIHandlerImpl$WorkbenchHelper.createPlatformResourceInputStream(PlatformResourceURIHandlerImpl.java:207)
	at org.eclipse.emf.ecore.resource.impl.PlatformResourceURIHandlerImpl.createInputStream(PlatformResourceURIHandlerImpl.java:525)
	at org.eclipse.emf.ecore.resource.impl.ExtensibleURIConverterImpl.createInputStream(ExtensibleURIConverterImpl.java:360)
	at org.eclipse.xtext.resource.XtextResourceSet$1.createInputStream(XtextResourceSet.java:298)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.load(ResourceImpl.java:1269)
	at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.demandLoad(ResourceSetImpl.java:259)
	at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.demandLoadHelper(ResourceSetImpl.java:274)
	at org.eclipse.xtext.resource.XtextResourceSet.getResource(XtextResourceSet.java:265)
	at org.eclipse.xtext.resource.SynchronizedXtextResourceSet.getResource(SynchronizedXtextResourceSet.java:25)
	at org.eclipse.xtext.ui.tests.resource.AbstractScopeResourceDescriptionsTest.assertDirtyState(AbstractScopeResourceDescriptionsTest.java:92)
	at org.eclipse.xtext.ui.tests.resource.PersistentStateScopeResourceDescriptionsTest.testDirtyStateAware(PersistentStateScopeResourceDescriptionsTest.java:31)

testLiveScope(org.eclipse.xtext.ui.tests.resource.PersistentStateScopeResourceDescriptionsTest)  Time elapsed: 0.026 sec
```

**Comment by cdietrich** [comment]
@szarnekow do you remember what this code is good for

```
		try {
			assertTrue(dirtyStateManager.manageDirtyState(mockDirtyResource));
			Resource dirtyResource = resourceSet.getResource(dirtyResourceURI, true);
			assertTrue(dirtyResource instanceof XtextResource);
			assertTrue(dirtyResource.isLoaded());
			assertFalse(dirtyResource.getContents().isEmpty());
			ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
			dirtyResource.save(byteArrayOutputStream, null);
			assertEquals(testModel, new String(byteArrayOutputStream.toByteArray()));
			assertTrue(enabled);
		} catch (Throwable t) {
			assertFalse(enabled);
			if (!(Throwables.getRootCause(t) instanceof org.eclipse.core.internal.resources.ResourceException))
				Throwables.propagate(t);
		} finally {
			dirtyStateManager.discardDirtyState(mockDirtyResource);
		}
	}
```

(the if (!(Throwables.getRootCause(t) instanceof org.eclipse.core.internal.resources.ResourceException))
				Throwables.propagate(t);)

in photon the root cause is a 

Throwables.getRootCause(t)
	 (java.lang.Exception) java.lang.Exception: Resource '/test/test.testlanguage' does not exist.

i am not sure what this test tests if the exception is always thrown....

**Comment by szarnekow** [comment]
Yes, that looks like a proper fix for the problem.

**Comment by cdietrich** [comment]
Fixed in 2.14
